### PR TITLE
Submitting Party for Staff Transfers 

### PR DESCRIPTION
**Pull request by dimak1**
*Issue #:* 
- bcgov/entity#15133

*Description of changes:*
- Add editable Submitting Party component to Review Screen (for Staff only)
- Add unit tests
- MHR Registration label is added because there was some refactoring done for Registration flow

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by cameron-eyds — APPROVED** [review]
LGTM!

**Review by doug-lovett — APPROVED** [review]
Looks good.

File: ppr-ui/src/store/actions/mhr-transfer-actions-model.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ export const setMhrTransferOwnLand: ActionIF = ({ commit }, isOwnLand: boolean):
   commit('mutateMhrTransferOwnLand', isOwnLand)
 }
 
+export const setMhrTransferSubmittingPartyKey: ActionIF = ({ commit }, { key, value }): void => {
+  commit('mutateMhrTransferSubmittingPartyKey', { key, value })
+}
+
```
Comment (cameron-eyds) [inline ppr-ui/src/store/actions/mhr-transfer-actions-model.ts:42]: Just seeing this external actions file now. 
I think if you were going to create this additional action file specific for Transfers, you should refactor all the actions specific to Transfers, not have a few here and a few there. It's pretty confusing. 
The action name also implies that you are setting a key

**Review by owillborn — APPROVED** [review]
LGTM

File: ppr-ui/src/store/mutations/mhr-transfer-mutations-model.ts [hunk 2]
```diff
@@ -16,6 +17,10 @@ export const mutateMhrTransferOwnLand = (state: StateIF, isOwnLand: boolean) =>
   state.stateModel.mhrTransfer.ownLand = isOwnLand
 }
 
+export const mutateMhrTransferSubmittingPartyKey = (state: StateIF, { key, value }) => {
```
Comment (cameron-eyds) [inline ppr-ui/src/store/mutations/mhr-transfer-mutations-model.ts:40]: Same as above with regards to the name and organization

File: ppr-ui/src/store/actions/mhr-transfer-actions-model.ts (hunk 1 above)
Comment (dimak1) [reply ppr-ui/src/store/actions/mhr-transfer-actions-model.ts:42]: Yea, the intention was to separate actions as the original file started to grow. I tried to keep all the Transfers-related actions in this file. Let me take a look if anything could be moved here, thanks!

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 3]
```diff
@@ -468,7 +496,8 @@ export default defineComponent({
       await parseMhrLocationInfo(currentLocationInfo)
 
       // Store existing submitting party to be used if user is BC Registry Staff
-      setMhrTransferSubmittingParty(data.submittingParty)
+      // setMhrTransferSubmittingParty(data.submittingParty)
+      // console.log('Mhr Info: ', data)
```
Comment (cameron-eyds) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue]: dev code

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 4]
```diff
@@ -524,31 +553,15 @@ export default defineComponent({
...
-          localState.accountInfo.name = getMhrTransferSubmittingParty.value.businessName
-        }
-        localState.accountInfo.accountAdmin = {
-          firstName: getMhrTransferSubmittingParty.value.personName?.firstName,
-          lastName: getMhrTransferSubmittingParty.value.personName?.LastName,
-          email: getMhrTransferSubmittingParty.value.emailAddress,
-          phone: getMhrTransferSubmittingParty.value.phoneNumber,
-          phoneExtension: getMhrTransferSubmittingParty.value.phoneExtension
-        }
-      }
```
Comment (cameron-eyds) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue:551]: Thank you

**Review by cameron-eyds — APPROVED** [review]
LGTM! 
Only question is our organization about the external file for actions/mutations specific to Transfers. 

While not opposed to the refactor, for consistency we should refactor all that apply. We now have transfer actions/mutations spread between multiple files and it's a bit confusing.

File: ppr-ui/src/store/actions/mhr-transfer-actions-model.ts (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/store/actions/mhr-transfer-actions-model.ts:42]: A good intention for sure, i think we just need to take it all the way to clear up any questions for future development.

Comment (dimak1) [reply ppr-ui/src/store/actions/mhr-transfer-actions-model.ts:42]: Re-organized the Transfer actions.

File: ppr-ui/src/store/mutations/mhr-transfer-mutations-model.ts (hunk 2 above)
Comment (dimak1) [reply ppr-ui/src/store/mutations/mhr-transfer-mutations-model.ts:40]: Re-organized the Transfer actions

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 3 above)
Comment (dimak1) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Cleaned up.

**Pull request merged**